amqtt/mqtt/protocol/handler.py

```diff
@@ -351,21 +352,25 @@ async def _handle_qos2_message_flow(self, app_message):
                     raise AMQTTException(message)
                 waiter = asyncio.Future()
                 self._pubrec_waiters[app_message.packet_id] = waiter
-                await waiter
-                del self._pubrec_waiters[app_message.packet_id]
-                app_message.pubrec_packet = waiter.result()
+                try:
+                    await waiter
+                    app_message.pubrec_packet = waiter.result()
```

Comment:
```suggestion
                    app_message.pubrec_packet = await waiter
```

Reply:
Fair call, was keeping the existing paradigm as the original. But you are right. It was set that way in order to delete the waiters in between